frontend/src/models/affinities.js

```diff
@@ -0,0 +1,74 @@
+export const affinityEvents = {
+  announcement: "announcement",
+  queen_of_clubs: "queen_of_clubs",
+  played_card: "played_card"
+};
+
+export class Affinities {
+  constructor(me, players = []) {
+    this.me = me;
+    this.setPlayers(players);
+  }
+
+  setPlayers(players) {
+    this.players = [];
+    players.forEach(player => {
+      if (this.me.id !== player.id) {
+        this.players.push({ player, affinity: 0 });
+      }
+    });
+  }
+
+  declaresParty(player) {
```

Comment:
The player model does already hold the party definition. I don't see a point why it should be theoretically possible to announce kontra when you're on the re team and therefore adding additionaly checks.

Since this is javascript, everything is on the client anyway, the door is anyway open for cheating.

I might be missing some security issues here, but as I initially though about this, I couldn't see a good reason for doing this.

I wasn't sure about the naming, but I believe if we stick to the way it is, my naming is more appropriate.
If it's getting changed, we should use yours 😃